docs/source/explanation/architecture-and-implementation.md

```diff
@@ -0,0 +1,67 @@
+(architecture-and-implementation)=
+
+# Architecture and Implementation
+
+## Architecture
+
+The `binderhub-service` chart runs the [BinderHub] Python software, in [api-only mode](https://binderhub.readthedocs.io/en/latest/reference/app.html#binderhub.app.BinderHub.enable_api_only_mode) (the default), as a standalone service to build, and push [Docker] images from source code repositories, on demand, using [repo2docker]. This service can then be paired with [JupyterHub] to allow users to initiate build requests from their hubs.
+
+Thus, the architecture of this system must:
+
+- facilitate the building and pushing of Docker images with repo2docker
+- easily integrate with a JupyterHub deployment
+- but also run as a standalone service
+- operate within a Kubernetes environment
```

Comment:
This heading includes a brief description, and requirements of the architecture.

In "Implementation details" below, we have:
- an image providing an overview of various pieces
- a text describing how pieces work together, to a large extent introduced on the spot

I'm thinking there could be benefits of a structure more like...

- Architecture requirements
  (Thus, the architecture of this system must...)
- Architecture overview
  (The great image you've created!)
- Details on how it works
  (When a build & push request is fired, the following events happen...)

By having such structure, I think we better decompose this documentation to be easier to expand and detail further. I reason that an "Architecture overview" for example could help us not need to introduce parts in other sections, because they are already introduced in an overview.

I think I've come to a meta conclusion: if structure we create easily can be used to describe parts in isolation - then the easier we can reference such parts without needing to duplicate and inline such notes elsewhere, and this in turn I think may lead to better being able to compose our documentation to be concise, easy to read, and maintain.

Reply:
Thanks @consideRatio! I like your proposed structure a lot and just added a commit to use it. Thnak you ✨